File: frontend/lib/api/coi.ts
// COI (Conflict of Interest) API Client
// Real backend integration (replaces coi-mock.ts)

import { apiFetch } from "@/lib/api/client"

// ============================================================================
// TYPES
// ============================================================================

export interface COIDashboardStats {
  conference_id: number
  total_reviewers: number
  available_reviewers: number
  total_papers: number
  papers_under_review: number
  coi_detected: number
  total_relationships: number
  total_assignments: number
  completed_assignments: number
}

export interface COIRelationship {
  id: number
  conference_id: number
  reviewer_id: number
  reviewer_name: string
  reviewer_email: string
  author_email: string // Primary identifier (not author_id)
  author_name: string
  author_affiliation?: string
  submission_id?: number
  type: string // "co_author" | "same_organization" | "advisor_advisee" | etc.
  severity: "high" | "medium" | "low"
  description: string
  evidence?: string[]
  start_date?: string
  end_date?: string
  detected_by: string
  paper_titles?: string[]
  created_at: string
  updated_at: string
}

export interface COIReport {
  reviewer_id: number
  reviewer_name: string
  reviewer_email: string
  reviewer_affiliation: string
  author_email: string // Primary identifier (not author_id)
  author_name: string
  author_affiliation: string
  coi_type: string
  severity: "high" | "medium" | "low" | "none"
  relationships: COIRelationship[]
  summary: string
  recommendation: "assign" | "review" | "avoid"
}

export interface AuthorInfo {
  email: string
  name: string
  affiliation?: string
}

export interface ConflictedReviewerInfo {
  reviewer_id: number
  reviewer_name: string
  reviewer_email: string
  severity: string
  reasons: string[]
}

export interface PaperCOISummary {
  paper_id: string
  paper_title: string
  authors: AuthorInfo[]
  total_conflicts: number
  high_severity_count: number
  medium_severity_count: number
  low_severity_count: number
  conflicted_reviewers: ConflictedReviewerInfo[]
}

// ============================================================================
// API FUNCTIONS
// ============================================================================

/**
 * Get COI dashboard statistics for a conference
 * Endpoint: GET /api/v1/coi/dashboard/stats/:conference_id
 */
export async function getCOIDashboardStats(
  conferenceId: number,
): Promise<COIDashboardStats> {
  const { data } = await apiFetch<{ data: COIDashboardStats }>(
    `/api/v1/coi/dashboard/stats/${conferenceId}`,
  )
  return data.data
}

/**
 * Get all COI relationships with filters and pagination
 * Endpoint: GET /api/v1/coi/relationships
 */
export async function getAllCOIRelationships(params: {
  conference_id: number // Required!
  severity?: "high" | "medium" | "low"
  relationship_type?: string
  search?: string
  limit?: number
  page?: number
}): Promise<{
  relationships: COIRelationship[]
  total: number
  page: number
  limit: number
}> {
  const queryParams = new URLSearchParams()
  queryParams.append("conference_id", params.conference_id.toString())
  if (params.severity) queryParams.append("severity", params.severity)
  if (params.relationship_type) queryParams.append("relationship_type", params.relationship_type)
  if (params.search) queryParams.append("search", params.search)
  if (params.limit) queryParams.append("limit", params.limit.toString())
  if (params.page) queryParams.append("page", params.page.toString())

  const { data } = await apiFetch<{ data: {
    relationships: COIRelationship[]
    total: number
    page: number
    limit: number
  } }>(`/api/v1/coi/relationships?${queryParams.toString()}`)
  
  return data.data
}

/**
 * Check detailed COI between a reviewer and an author
 * Endpoint: GET /api/v1/coi/check/reviewer/:reviewer_id/author/:author_email
 * Note: Uses author_email, not author_id!
 */
export async function checkReviewerToAuthorCOI(
  conferenceId: number,
  reviewerId: number,
  authorEmail: string,
): Promise<COIReport> {
  const { data } = await apiFetch<{ data: COIReport }>(
    `/api/v1/coi/check/reviewer/${reviewerId}/author/${encodeURIComponent(authorEmail)}?conference_id=${conferenceId}`,
  )
  return data.data
}

/**
 * Get COI summaries for all papers in a conference
 * Endpoint: GET /api/v1/coi/papers
 */
export async function getAllPaperCOIs(params: {
  conference_id: number // Required!
  severity?: "high" | "medium" | "low"
  search?: string
  limit?: number
  page?: number
}): Promise<{
  papers: PaperCOISummary[]
  total: number
  page: number
  limit: number
}> {
  const queryParams = new URLSearchParams()
  queryParams.append("conference_id", params.conference_id.toString())
  if (params.severity) queryParams.append("severity", params.severity)
  if (params.search) queryParams.append("search", params.search)
  if (params.limit) queryParams.append("limit", params.limit.toString())
  if (params.page) queryParams.append("page", params.page.toString())

  const { data } = await apiFetch<{ data: {
    papers: PaperCOISummary[]
    total: number
    page: number
    limit: number
  } }>(`/api/v1/coi/papers?${queryParams.toString()}`)
  
  return data.data
}

/**
 * Rebuild COI relationships for a conference
 * Endpoint: POST /api/v1/coi/conferences/:conference_id/rebuild
 * Admin/Chair only
 */
export async function rebuildCOIRelationships(
  conferenceId: number,
): Promise<{
  conference_id: number
  relationships_found: number
  relationships_stored: number
  detection_time_ms: number
}> {
  const { data } = await apiFetch<{ data: any }>(
    `/api/v1/coi/conferences/${conferenceId}/rebuild`,
    {
      method: "POST",
    },
  )
  return data.data
}

File: backend/internal/dto/coi.go
package dto

import (
	"encoding/json"
	"time"
)

// ============================================================================
// COI Relationship DTOs
// ============================================================================

// COIRelationship represents a conflict of interest relationship with enriched details
type COIRelationship struct {
	ID               int64      `json:"id"`
	ConferenceID     int64      `json:"conference_id"`
	ReviewerID       int64      `json:"reviewer_id"`
	ReviewerName     string     `json:"reviewer_name"`
	ReviewerEmail    string     `json:"reviewer_email"`
	AuthorEmail      string     `json:"author_email"`
	AuthorName       string     `json:"author_name"`
	AuthorAffiliation string    `json:"author_affiliation,omitempty"`
	SubmissionID     *int64     `json:"submission_id,omitempty"`
	RelationshipType string     `json:"type"`
	Severity         string     `json:"severity"`
	Description      string     `json:"description"`
	Evidence         []string   `json:"evidence,omitempty"`
	StartDate        *time.Time `json:"start_date,omitempty"`
	EndDate          *time.Time `json:"end_date,omitempty"`
	DetectedBy       string     `json:"detected_by"`
	PaperTitles      []string   `json:"paper_titles,omitempty"`
	CreatedAt        time.Time  `json:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at"`

	// Internal field for JSONB evidence
	EvidenceJSON []byte `json:"-"`
}

// UnmarshalEvidence parses the JSONB evidence field
func (c *COIRelationship) UnmarshalEvidence() error {
	if len(c.EvidenceJSON) > 0 {
		return json.Unmarshal(c.EvidenceJSON, &c.Evidence)
	}
	c.Evidence = []string{}
	return nil
}

// COIRelationshipListRequest represents query parameters for listing relationships
type COIRelationshipListRequest struct {
	ConferenceID     int64  `form:"conference_id" binding:"required"`
	Severity         string `form:"severity"`
	RelationshipType string `form:"relationship_type"`
	Search           string `form:"search"`
	Limit            int    `form:"limit"`
	Page             int    `form:"page"`
}

// COIRelationshipListResponse represents the paginated response for relationships
type COIRelationshipListResponse struct {
	Relationships []*COIRelationship `json:"relationships"`
	Total         int64              `json:"total"`
	Page          int                `json:"page"`
	Limit         int                `json:"limit"`
}

// ============================================================================
// Dashboard Stats DTOs
// ============================================================================

// COIDashboardStats represents statistics for the COI dashboard
type COIDashboardStats struct {
	ConferenceID          int64 `json:"conference_id"`
	TotalReviewers        int   `json:"total_reviewers"`
	AvailableReviewers    int   `json:"available_reviewers"`
	TotalPapers           int   `json:"total_papers"`
	PapersUnderReview     int   `json:"papers_under_review"`
	COIDetected           int   `json:"coi_detected"`
	TotalRelationships    int   `json:"total_relationships"`
	TotalAssignments      int   `json:"total_assignments"`
	CompletedAssignments  int   `json:"completed_assignments"`
}

// COIDashboardStatsRequest represents the request for dashboard stats
type COIDashboardStatsRequest struct {
	ConferenceID int64 `uri:"conference_id" binding:"required"`
}

// ============================================================================
// COI Report DTOs (Detailed Check)
// ============================================================================

// COIReport represents a detailed COI analysis for a reviewer-author pair
type COIReport struct {
	ReviewerID          int64               `json:"reviewer_id"`
	ReviewerName        string              `json:"reviewer_name"`
	ReviewerEmail       string              `json:"reviewer_email"`
	ReviewerAffiliation string              `json:"reviewer_affiliation"`
	AuthorEmail         string              `json:"author_email"`
	AuthorName          string              `json:"author_name"`
	AuthorAffiliation   string              `json:"author_affiliation"`
	COIType             string              `json:"coi_type"` // "author"
	Severity            string              `json:"severity"`
	Relationships       []*COIRelationship  `json:"relationships"`
	Summary             string              `json:"summary"`
	Recommendation      string              `json:"recommendation"` // "assign", "review", "avoid"
}

// COICheckRequest represents the request for checking reviewer-author COI
type COICheckRequest struct {
	ReviewerID  int64  `uri:"reviewer_id" binding:"required"`
	AuthorEmail string `uri:"author_email" binding:"required"`
}

// ============================================================================
// Paper COI Summary DTOs
// ============================================================================

// PaperCOISummary represents COI summary for a single paper
type PaperCOISummary struct {
	PaperID              string                   `json:"paper_id"`
	PaperTitle           string                   `json:"paper_title"`
	Authors              []*AuthorInfo            `json:"authors"`
	TotalConflicts       int                      `json:"total_conflicts"`
	HighSeverityCount    int                      `json:"high_severity_count"`
	MediumSeverityCount  int                      `json:"medium_severity_count"`
	LowSeverityCount     int                      `json:"low_severity_count"`
	ConflictedReviewers  []*ConflictedReviewerInfo `json:"conflicted_reviewers"`
}

// AuthorInfo represents basic author information
type AuthorInfo struct {
	Email       string `json:"email"`
	Name        string `json:"name"`
	Affiliation string `json:"affiliation,omitempty"`
}

// ConflictedReviewerInfo represents a reviewer with conflicts for a paper
type ConflictedReviewerInfo struct {
	ReviewerID    int64    `json:"reviewer_id"`
	ReviewerName  string   `json:"reviewer_name"`
	ReviewerEmail string   `json:"reviewer_email"`
	Severity      string   `json:"severity"`
	Reasons       []string `json:"reasons"`
}

// PaperCOIListRequest represents query parameters for listing paper COIs
type PaperCOIListRequest struct {
	ConferenceID int64  `form:"conference_id" binding:"required"`
	Severity     string `form:"severity"`
	Search       string `form:"search"`
	Limit        int    `form:"limit"`
	Page         int    `form:"page"`
}

// PaperCOIListResponse represents the paginated response for paper COIs
type PaperCOIListResponse struct {
	Papers []*PaperCOISummary `json:"papers"`
	Total  int64              `json:"total"`
	Page   int                `json:"page"`
	Limit  int                `json:"limit"`
}

// ============================================================================
// COI Rebuild Request (Admin)
// ============================================================================

// COIRebuildRequest represents the request to rebuild COI relationships
type COIRebuildRequest struct {
	ConferenceID int64 `uri:"conference_id" binding:"required"`
}

// COIRebuildResponse represents the response after rebuilding COI relationships
type COIRebuildResponse struct {
	ConferenceID        int64 `json:"conference_id"`
	RelationshipsFound  int   `json:"relationships_found"`
	RelationshipsStored int   `json:"relationships_stored"`
	DetectionTimeMs     int64 `json:"detection_time_ms"`
}





